vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3295,4 +3349,225 @@ private void handleDataProviderChange(DataProvider<T, ?> dataProvider) {
+    }
+
+    /**
+     * Sets the drop mode of this drop target.
+     * <p>
+     * When using {@link DropMode#ON_TOP}, and the grid is either empty or has
+     * empty space after the last row, the drop can still happen on the empty
+     * space, and the {@link GridDropEvent#getDropTargetRow()} will return an
+     * empty optional.
+     * <p>
+     * When using {@link DropMode#BETWEEN} or
+     * {@link DropMode#ON_TOP_OR_BETWEEN}, and there is at least one row in the
+     * grid, any drop after the last row in the grid will get the last row as
+     * the {@link GridDropEvent#getDropTargetRow()}. If there are no rows in the
+     * grid, then it will return an empty optional.
+     * <p>
+     * If using {@link DropMode#ON_GRID}, then the drop will not happen on any
+     * row, but instead just "on the grid". The target row will not be present
+     * in this case.
+     * <p>
+     * <em>NOTE: Prefer not using a row specific {@link DropMode} with a grid
+     * that enables sorting. If for example a new row gets added to a specific
+     * location on drop event, it might not end up in the location of the drop
+     * but rather where the active sorting configuration prefers to place it.
+     * This behavior might feel unexpected for the users.
+     * 
+     * @param dropMode
+     *            Drop mode that describes the allowed drop locations within the
+     *            Grid's row.
+     * @see GridDropEvent#getDropLocation()
+     */
+    public void setDropMode(DropMode dropMode) {
+        getElement().setProperty("dropMode",
+                dropMode == null ? null : dropMode.getClientName());
+    }
+
+    /**
+     * Gets the drop mode of this drop target.
+     *
+     * @return Drop mode that describes the allowed drop locations within the
+     *         Grid's row.
+     */
+    public DropMode getDropMode() {
+        String dropMode = getElement().getProperty("dropMode");
+        Optional<DropMode> mode = Arrays.stream(DropMode.values())
+                .filter(dm -> dm.getClientName().equals(dropMode)).findFirst();
+        return mode.orElse(null);
+    }
+
+    /**
+     * Sets whether the user can drag the grid rows or not.
+     *
+     * @param rowsRraggable
+     *            {@code true} if the rows can be dragged by the user;
+     *            {@code false} if not
+     */
+    public void setRowsDraggable(boolean rowsRraggable) {
+        getElement().setProperty("rowsDraggable", rowsRraggable);
+    }
+
+    /**
+     * Gets whether rows of the grid can be dragged.
+     *
+     * @return {@code true} if the rows are draggable, {@code false} otherwise
+     */
+    public boolean isRowsDraggable() {
+        return getElement().getProperty("rowsDraggable", false);
+    }
+
+    /**
+     * Gets the active drop filter.
+     * 
+     * @return The drop filter function
+     */
+    public SerializableFunction<T, Boolean> getDropFilter() {
+        return this.dropFilter;
```

Comment:
The usual convention is not to use `this`. Not blocking

Reply:
Done.